lib/inspec/transport.rb

```diff
@@ -0,0 +1,78 @@
+# encoding: utf-8
+
+require 'train'
+
+module Inspec
+  class Transport
+    attr_accessor :cache_resources, :conn
+    include Inspec::Backend::Base
+
+    def initialize(config)
+      @cache_resources = false
+      @cmd_cache = {}
+      @file_cache = {}
+      init_train(config)
+    end
+
+    def init_train(config)
+      conf = Train.target_config(config)
+      name = Train.validate_backend(conf)
+      transport = Train.create(name, conf)
+
+      if transport.nil?
+        raise "Can't find transport backend '#{name}'."
+      end
+
+      @conn = transport.connection
+      if @conn.nil?
+        raise "Can't connect to transport backend '#{name}'."
+      end
+    rescue Train::ClientError => e
+      raise "Client error, can't connect to '#{name}' backend: #{e.message}"
+    rescue Train::TransportError => e
+      raise "Transport error, can't connect to '#{name}' backend: #{e.message}"
+    end
+
+    def clear_cache
+      @cmd_cache = {}
+      @file_cache = {}
+    end
+
+    def os
+      @conn.os
+    end
+    alias platform os
+
+    # This is needed for the mock connection which has some
+    # unique method calls
```

Comment:
Can we provide an example or two in the comment as to the type of method calls we're catching here, deeming this necessary?